### PR TITLE
Implement IN string hole.

### PR DESCRIPTION
**Pull request by Wosin**

### CODE REVIEW

File: src/main/resources/com/nec/arrow/functions/cpp/frovedis/text/dict.cc [hunk 1]
```diff
@@ -654,11 +654,11 @@ std::vector<size_t> dict::lookup(const compressed_words& cw) const {
   std::vector<size_t> ret(num_words_lookup);
   auto retp = ret.data();
 
-  size_t left[DICT_VLEN]; // need to be signed?
+  ssize_t left[DICT_VLEN]; // need to be signed?
```
Comment (wgip) [inline src/main/resources/com/nec/arrow/functions/cpp/frovedis/text/dict.cc:657]: What is the difference here?  cc @q10

File: src/main/scala/com/nec/spark/agile/StringHole.scala [hunk 2]
```diff
@@ -60,6 +60,42 @@ object StringHole {
...
+        StringHoleEvaluation {
+      val valuesWords = s"in_values_${Math.abs(hashCode())}"
+      val toCheckWords = s"in_toCheck_${Math.abs(hashCode())}"
+      val filteredIds = s"in_filtered_${Math.abs(hashCode())}"
+      val matchingIds = s"in_matches_${Math.abs(hashCode())}"
+
+      val words = valueList.mkString(" ").map(_.toInt).mkString(",")
+
+      override def computeVector: CodeLines = CodeLines.from(
+    s"frovedis::words $valuesWords = varchar_vector_to_words($refName);",
```
Comment (wgip) [inline src/main/scala/com/nec/spark/agile/StringHole.scala:73]: run scalafmt on this file

File: src/test/scala/com/nec/cmake/eval/InStringHoleEvaluationSpec.scala [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+    val results = executeHoleEvaluation(list, evaluation)
+
+    assert(results == expectedResults)
+  }
+
+  "It" should  "correctly filter out input set if no matches are preset" in {
+    val list = List("Dog", "Cat", "Cow", "Hotel", "Cyclone", "Spark", "Brown", "Fox")
+    val toMatchList = List("not", "here","Has")
+    val expectedResults = list.collect{
```
Comment (wgip) [inline src/test/scala/com/nec/cmake/eval/InStringHoleEvaluationSpec.scala:34]: nice 👍 
plus format the file

**Review by wgip — APPROVED** [review]
just need a bit of formatting but looks great

Which queries does this @Wosin ?

**Comment by Wosin** [comment]
Q12

> just need a bit of formatting but looks great
> 
> Which queries does this fix @Wosin?

It fixes Q12, not suree if that was happening in any other query.

**Pull request merged**